tendrl/gluster_integration/sds_sync/utilization.py

```diff
@@ -39,28 +39,72 @@ def sync_utilization_details(volumes):
     )
     out = ''
     try:
+        # 'out' is a dictonary with keys 'error' and 'success', each of
+        # which holds 'volume' and it's 'utilization' or 'error' respectively
+        # as key-value pairs.
+        # {
+        # "success": {
+        #     "volume1": {"total": 3243243, "used": 45435435, ...},
+        #     "volume2": {"total": 3243243, "used": 45435435, ...},
+        #     ....
+        # },
+        # "error": {
+        #     "volume3": "error message",
+        #     "volume4": "error message",
+        #     ....
+        # }
+        # }
+        # 'err' stores generic stderr messages for logging purpose
         out, err = cmd.communicate()
+
         util_det = json.loads(out)
-        for k, v in util_det.iteritems():
-            volume = voldict[k]
-            volume.usable_capacity = int(v['total'])
-            volume.used_capacity = int(v['used'])
-            volume.pcnt_used = str(v['pcnt_used'])
-            volume.total_inode_capacity = int(v['total_inode'])
-            volume.used_inode_capacity = int(v['used_inode'])
-            volume.pcnt_inode_used = str(v['pcnt_inode_used'])
-            volume.save()
-            cluster_used_capacity += volume.used_capacity
-            cluster_usable_capacity += volume.usable_capacity
-    except(KeyError, TypeError, ValueError, AttributeError):
+        if err != '':
+            logger.log(
+                "error",
+                NS.publisher_id,
+                {'message': 'Error : %s\n' % err}
+            )
+        if util_det:
+            if 'error' in util_det.keys():
+                logger.log(
+                    "error",
+                    NS.publisher_id,
+                    {'message': 'Error getting utilization of some '
+                     'volumes\nError: %s\n' % util_det['error']}
+                )
+
+            if 'success' in util_det.keys():
```

Comment:
will work but .keys() is not required.

Reply:
it is better to check with .keys(), because here success is a key only, so checking with .keys() is actually safe